src/status_im2/common/animated_header_list/view.cljs

```diff
@@ -0,0 +1,101 @@
+(ns status-im2.common.animated-header-list.view
+  (:require
+   [quo2.core :as quo]
+   [react-native.core :as rn]
+   [react-native.platform :as platform]
+   [react-native.reanimated :as reanimated]
+   [react-native.safe-area :as safe-area]
+   [reagent.core :as reagent]
+   [quo2.foundations.colors :as colors]
+   [status-im.ui.components.fast-image :as fast-image]
+   [status-im2.common.animated-header-list.style :as style]
+   [oops.core :as oops]
+   [utils.re-frame :as rf]))
+
+(def header-height 192)
+
+(def threshold (/ header-height 2))
+
+(defn scroll-handler
+  [event initial-y scroll-y opacity-value]
+  (let [current-y (- (oops/oget event "nativeEvent.contentOffset.y") initial-y)]
+    (reanimated/set-shared-value scroll-y current-y)
+    (reanimated/set-shared-value opacity-value
+                                 (reanimated/with-timing (if (> current-y threshold) 1 0)))))
```

Comment:
@OmarBasem, I'm wondering if we can avoid the repeated calls to `set-shared-value` to set the opacity.

For instance, there will be many different `ys` that will be greater than `threshold`, but all of them will set the value to `1`. Instead, I think we can optimize to only set the shared value if it's different than the previous one. I would say it's worth a shot.

```clojure
(defn scroll-handler
  [event initial-y scroll-y opacity-value]
  (let [current-y   (- (oops/oget event "nativeEvent.contentOffset.y") initial-y)
        new-opacity (if (> current-y threshold) 1 0)]
    (reanimated/set-shared-value scroll-y current-y)
    (when (not= new-opacity opacity-value)
      (reanimated/set-shared-value opacity-value
                                   (reanimated/with-timing new-opacity)))))
```

Reply:
I thought about this, but I believe reanimated is going to ignore setting the same value. But I will check and confirm.

Reply:
Got rid of it altogether